[Notifier] add doc (#12846)
symfony/symfony-docs: components/notifier.rst

.. index::
   single: Notifier
   single: Components; Notifier

The Notifier Component
====================

>     The Notifier component helps sending emails.

Reviewer: ```suggestion
    The Notifier component helps sending notifications (emails, Slack, SMS).
```

If you're using the Symfony Framework, read the
:doc:`Symfony Framework Notifier documentation </notifier>`.

Installation
------------

.. code-block:: terminal

    $ composer require symfony/notifier

.. include:: /components/require_autoload.rst.inc

Email
-----

> The Notifier component has notify you when something goes wrong::

Reviewer: ```suggestion
The Notifier component can notify someone(s) when something occurs::
```

it can be you or someone else, that the devops or whomever
also, not only when goes wrong, it can be something good happening :)

>     use Symfony\Bridge\Twig\Mime\NotificationEmail;

Reviewer: generaly the doc use to put such code in context, inside a controller or a service for example
can you reproduce the same? i think it will be very usefull :)

    use Symfony\Component\Mailer\Mailer;
    use Symfony\Component\Mailer\Transport\Smtp\EsmtpTransport;

    $transport = new EsmtpTransport('localhost');
    $mailer = new Mailer($transport);

    $email = (new NotificationEmail())
        ->form('fabien@symfony.com')
        ->to('fabien@symfony.com')
        ->exception($exception);

    $mailer->send($email);

The ``$email`` object is created via the :doc:`Mime component </components/mime>`.

And configurable email template::

    $email = (new NotificationEmail())
        ->htmlEmail('email/system.html.twig')
        ->textEmail('email/system.txt.twig');

With template::

    {% extends "@email/system.html.twig" %}
    {% block style %}
        {{ parent() }}
        .container.body_alert {
            border-top: 30px solid #ec5840;
        }
    {% endblock %}
    {% block lines %}

>         This is an automated email for the MyApp application.

Reviewer: ```suggestion
        This is a notification email for the MyApp application.
```

        {{ parent() }}
    {% endblock %}
    {% block action %}
        {{ parent() }}
        <spacer size="16"></spacer>
        <button class="secondary" href="https://myapp.com/">Go to MyApp</button>
    {% endblock %}
    {% block exception %}{% endblock %}
    {% block footer_content %}
        <p><small>&copy; MyApp</small></p>
    {% endblock %}

SMS
---------

> Sending SMS Messages the easy way::

Reviewer: SMS stands for Short Message Service
so Short Message Service Messages seems weird, what about

```suggestion
Sending SMS notifications the easy way::
```

?

    /**
    * @Route("/checkout/thankyou")
    */
    public function thankyou(Texter $texter /* ... */) {

>         $sms = new SmsMessage('+1415999888', 'Revenue has just increased by 1€ per year!');

Reviewer: need the `use` refering to this SmsMessage
